src/glat.cc

```diff
@@ -201,10 +201,11 @@ bool OpenTypeGLAT_v3::Parse(const uint8_t* data, size_t length,
         return DropGraphite("Illegal nested compression");
       }
       std::vector<uint8_t> decompressed(this->compHead & FULL_SIZE, 0);
-      int ret = LZ4_decompress_safe(
+      int ret = LZ4_decompress_safe_partial(
           reinterpret_cast<const char*>(data + table.offset()),
           reinterpret_cast<char*>(decompressed.data()),
           table.remaining(),
+          decompressed.size(),
           decompressed.size());
       if (ret < 0) {
```

Comment:
I think rather than just checking for `ret < 0` here, we should explicitly check that `ret == decompressed.size()` to confirm that decompression generated the expected amount of data; if not, then even though LZ4 didn't report an error, something is still badly wrong.